tests/test_project.py

```diff
@@ -477,6 +477,323 @@ def test_update_some_with_imports(repos_tmpdir):
     manifest = Manifest.from_file(topdir=ws)
     assert manifest.get_projects(['Kconfiglib'])[0].is_cloned()
 
+def test_update_submodules_list(repos_tmpdir):
+    # The west update command should not only update projects,
+    # but also its submodules. Test uses two pairs of project
+    # and submodule checking out submodule in the default
+    # and custom location to verify both cases.
+
+    # Repositories paths
+    remotes = repos_tmpdir / 'repos'
+    # tagged_repo is zephyr's submodule and will be updated
+    # in the default location (i.e. zephyr/tagged_repo).
+    zephyr = remotes / 'zephyr'
+    tagged_repo = remotes / 'tagged_repo'
+    # Kconfiglib is net_tools's submodule and will be updated
+    # in the custom location (i.e. net-tools/third_parties/Kconfiglib).
+    net_tools = remotes / 'net-tools'
+    kconfiglib = remotes / 'Kconfiglib'
+    kconfiglib_submodule = 'third_parties/Kconfiglib'
+
+    # Creating west workspace and manifest repository.
+    ws = repos_tmpdir / 'ws'
+    create_workspace(ws, and_git=True)
+    manifest_repo = ws / 'mp'
+    create_repo(manifest_repo)
+
+    # Commit west.yml describing projects and submodules dependencies.
+    add_commit(manifest_repo, 'manifest repo commit',
+               files={'west.yml':
+                      f'''
+                      manifest:
+                        projects:
+                        - name: zephyr
+                          url: {zephyr}
+                          submodules:
+                            - name: tagged_repo
+                              path: tagged_repo
+                        - name: net-tools
+                          url: {net_tools}
+                          submodules:
+                            - name: Kconfiglib
+                              path: {kconfiglib_submodule}
```

Comment:
Plural starts at two - is there a test with at least 2 submodules in one west project? If not then there's lack of test coverage.

Reply:
The `test_update_all_submodules` test is having 2 submodules in one project and updates them, by assigning true value to the submodules. Do you think it covers mentioned use case, or it should be verified in some other test if passing at least two elements to the submodules list works?

Reply:
Thanks, I barely reviewed the test code and this was not the easiest to spot because as you wrote there's no _explicit_ list with more than 1 submodule in the test inputs.

> or it should be verified in some other test if passing at least two elements to the submodules list works?

I don't think that is necessary because that would add coverage only for the parsing / input parts of the code and west relies on YAML for most of that.